indexer/processor/l2_processor.go

```diff
@@ -235,6 +241,100 @@ func l2ProcessContractEventsBridgeTransactions(processLog log.Logger, db *databa
 	return nil
 }
 
+func l2ProcessContractEventsBridgeCrossDomainMessages(processLog log.Logger, db *database.DB, events *ProcessedContractEvents) error {
+	l2ToL1MessagePasserABI, err := bindings.NewL2ToL1MessagePasser(common.Address{}, nil)
+	if err != nil {
+		return err
+	}
+
+	// (2) Process New Messages
+	sentMessageEvents, err := CrossDomainMessengerSentMessageEvents(events)
+	if err != nil {
+		return err
+	}
+
+	sentMessages := make([]*database.L2BridgeMessage, len(sentMessageEvents))
+	for i, sentMessageEvent := range sentMessageEvents {
+		log := events.eventLog[sentMessageEvent.RawEvent.GUID]
+
+		// extract the withdrawal hash from the previous MessagePassed event
+		msgPassedLog := events.eventLog[events.eventByLogIndex[ProcessedContractEventLogIndexKey{log.BlockHash, log.Index - 1}].GUID]
+		msgPassedEvent, err := l2ToL1MessagePasserABI.ParseMessagePassed(*msgPassedLog)
+		if err != nil {
+			return err
+		}
+
+		sentMessages[i] = &database.L2BridgeMessage{
+			TransactionWithdrawalHash: msgPassedEvent.WithdrawalHash,
+			BridgeMessage: database.BridgeMessage{
+				Nonce:                database.U256{Int: sentMessageEvent.MessageNonce},
+				MessageHash:          sentMessageEvent.MessageHash,
+				SentMessageEventGUID: sentMessageEvent.RawEvent.GUID,
+				GasLimit:             database.U256{Int: sentMessageEvent.GasLimit},
+				Tx: database.Transaction{
+					FromAddress: sentMessageEvent.Sender,
+					ToAddress:   sentMessageEvent.Target,
+					Amount:      database.U256{Int: sentMessageEvent.Value},
+					Data:        sentMessageEvent.Message,
+					Timestamp:   sentMessageEvent.RawEvent.Timestamp,
+				},
+			},
+		}
+	}
+
+	if len(sentMessages) > 0 {
+		processLog.Info("detected L2CrossDomainMessenger messages", "size", len(sentMessages))
+		err := db.BridgeMessages.StoreL2BridgeMessages(sentMessages)
+		if err != nil {
+			return err
+		}
+	}
+
+	// (2) Process Relayed Messages.
+	//
+	// NOTE: Should we care about failed messages? A failed message can be
+	// inferred via an included deposit on L2 that has not been marked as relayed.
+	relayedMessageEvents, err := CrossDomainMessengerRelayedMessageEvents(events)
+	if err != nil {
+		return err
+	}
+
+	for _, relayedMessage := range relayedMessageEvents {
+		message, err := db.BridgeMessages.L1BridgeMessageByHash(relayedMessage.MsgHash)
+		if err != nil {
+			return err
+		} else if message == nil {
+			// Since the transaction processor running prior does not ensure the deposit inclusion, we need to
+			// ensure we are in a caught up state before claiming a missing event. Since L2 timestamps are derived
+			// from L1, we can simply compare the timestamp of this event with the latest L1 header.
+			latestL1Header, err := db.Blocks.LatestL1BlockHeader()
```

Comment:
Small corner case here possibly. I think if the blocktime of an OP chain is less than 1 second multiple blocks can have the same timestamp and thus this would fail to be true. Ponder was having issues related to this recently not sure if it affects OP chains

nit: would prefer comment to be on line 314

Reply:
synced offline. we dont have a <1s blocktime. Although we should explore any usecase where someone launches an opchain with this functionality

Once i rework the event tracking process to not rely on waiting for the opposite processor, this check will be removed anyways